File: JS/31-40/e_39.js
function sumaDigitos(numero) {
    // Inicializa la variable para almacenar la suma
    let suma = 0;

    // Convierte el número a su representación como cadena de caracteres
    const strNumero = String(numero);

    // Itera sobre cada dígito y suma su valor convertido a entero
    for (let i = 0; i < strNumero.length; i++) {
        suma += parseInt(strNumero[i], 10);
    }

    return suma;
}

// Solicita al usuario que ingrese un número
const numeroUsuario = parseInt(prompt("Ingresa un número:"), 10);

// Calcula la suma de los dígitos utilizando la función
const resultado = sumaDigitos(numeroUsuario);

// Imprime el resultado
console.log(`La suma de los dígitos de ${numeroUsuario} es: ${resultado}`);


// Este programa utiliza una función llamada sumaDigitos para calcular la suma de los dígitos del número ingresado por el usuario.
// Se solicita al usuario que ingrese un número mediante el uso de prompt y luego se imprime el resultado en la consola.
// La función parseInt se utiliza para convertir la entrada del usuario a un número entero.

File: Python/31-40/e_39.py
# Escribe en python un programa que calcule la suma de los dígitos de un número, imprime el resultado.


def suma_digitos(numero):
    # Inicializa la variable para almacenar la suma
    suma = 0
    
    # Convierte el número a su representación como cadena de caracteres
    str_numero = str(numero)
    
    # Itera sobre cada dígito y suma su valor convertido a entero
    for digito in str_numero:
        suma += int(digito)
    
    return suma

# Solicita al usuario que ingrese un número
numero_usuario = int(input("Ingresa un número: "))

# Calcula la suma de los dígitos utilizando la función
resultado = suma_digitos(numero_usuario)

# Imprime el resultado
print(f"La suma de los dígitos de {numero_usuario} es: {resultado}")


# Este programa solicita al usuario que ingrese un número, 
# luego utiliza una función suma_digitos para calcular la suma de los dígitos del número ingresado y finalmente imprime el resultado.

